ldp/nn/handlers/transformer_handler.py

```diff
@@ -0,0 +1,1017 @@
+                if isinstance(func, str)
+                else func(self, *args, **kwargs)
+            )
+
+        # Needed to prevent GPU memory leak to the main process scheduling the workers
+        if isinstance(res, GenerateDecoderOnlyOutput):
+            res.past_key_values = None
+            res["past_key_values"] = None
+
+        to_cpu = partial(_move_tensor, device=torch.device("cpu"))
+        return tree.map_structure(to_cpu, res)
+
+    def __del__(self) -> None:
+        dist.destroy_process_group()
+
+
+class ParallelAsyncTransformer(AsyncTransformerInterface):
+    def __init__(self, config: TransformerHandlerConfig):
+        self._initialized = False
+
+        parallel_mode_config = config.parallel_mode_config
+        if not parallel_mode_config:
+            raise ValueError("Parallel mode config must be provided.")
+        self.config = config
+        self.tokenizer = config.lm_config.get_tokenizer()
+        maybe_set_tokenizer_chat_template(self.tokenizer)
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+
+        match parallel_mode_config.execution_mode:
+            case ExecutionMode.LOCAL_MACHINE:
+                self._init_local_cluster(config, parallel_mode_config)
+            case ExecutionMode.SLURM_CLUSTER:
+                self._init_slurm_cluster(config, parallel_mode_config)
+            case _:
+                raise NotImplementedError(
+                    f"Unsupported execution mode: {parallel_mode_config.execution_mode}"
+                )
+
+        self._initialized = True
+
+        # don't call AsyncTorchModule.__init__ because we don't need to set up module[_call_fn]
+        AsyncBufferedWorker.__init__(
+            self,
+            batch_size=config.batch_size,
+            max_wait_interval=config.max_wait_interval,
+            collate_fn=config.collate_fn,
+            decollate_fn=config.decollate_fn,
+        )
+
+        def handler_call_fn(handler: ParallelTransformerHandler, *args, **kwargs):
+            return config.module_call_fn(handler.module, *args, **kwargs)
+
+        self.handler_call_fn = handler_call_fn
+
+    def _init_local_cluster(self, config, parallel_mode_config):
+        """Initialize a Dask cluster on local machine."""
+        try:
+            self.cluster = LocalCUDACluster(
+                n_workers=parallel_mode_config.num_workers,
+                threads_per_worker=parallel_mode_config.num_cpus_per_worker,
+                host=parallel_mode_config.scheduler_addr,
+                port=parallel_mode_config.scheduler_port,
+                memory_limit=None,  # do not let Dask manage memory - if we OOM, we OOM
+            )
+        except TypeError:
+            raise ImportError(
+                f"To run {type(self).__name__} with execution mode"
+                f" {parallel_mode_config.execution_mode}, please install the"
+                " 'dask-cuda' package by running `make nn` in the repo root."
```

Comment:
`make nn` is not a thing here, can you update this?

Reply:
I've removed the try/except altogether.